api/src/pcapi/scripts/fix_venues_oa/main.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import logging
+
+import sqlalchemy as sa
+
+from pcapi.app import app
+from pcapi.core.offerers.models import OffererAddress
+from pcapi.core.offerers.models import Venue
+from pcapi.core.offers.models import Offer
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+app.app_context().push()
+OFFER_BATCH_SIZE = 500
+
+
+def create_offerer_address_for_venues(dry_run: bool = True) -> None:
+    offerer_addresses_with_multiple_venues = (
+        OffererAddress.query.join(Venue, Venue.offererAddressId == OffererAddress.id)
+        .group_by(OffererAddress.id)
+        .having(db.func.count(Venue.id) > 1)
+        .options(sa.orm.selectinload(OffererAddress.venues))
+        .all()
+    )
```

Comment:
Tu fais un `selectinload` ici des venues, mais juste en dessous, on fait une subquery pour fetch les venues de chaque offerer, puis une query pour fetch les venue.name, venue.publicName et l'adresse de chaque offerer address. Je te propose une modification de cette query qui remplace les deux du dessous (le `selectinload` remplace déjà la subquery du dessous, c'est fait pour que l'orm charge toutes les venues à l'instant où tu fait `offerer_address.venues`)

```suggestion
    offerer_addresses_with_multiple_venues = (
        OffererAddress.query.join(Venue, Venue.offererAddressId == OffererAddress.id)
        .options(sa.orm.joinedload(Address))
        .group_by(OffererAddress.id)
        .having(db.func.count(Venue.id) > 1)
        .options(sa.orm.selectinload(OffererAddress.venues).load_only(Venue.id, Venue.name, Venue.publicName)
        .all()
    )
```

Reply:
my bad, je n'avais pas vu le count(offer)/order_by(offer) ci-dessous, tu peux laisser comme ça du coup. Par contre le `yield_per` pour la mise à jour risque d'être nécessaire sous peine de statement timeout